### PR TITLE
Triple adaptation and refactor demo

### PR DESCRIPTION
**Pull request by DMwangnima**
This PR introduces new Triple protocol based on Connect library without generic and Triple adaptation.
Main change could be seen in /consumer, /provider and /protocol/triple.
This is a POC.

### CODE REVIEW

**Comment by chickenlj** [comment]
Please run go fmt to fix the code format issue.

File: protocol/triple/internal/client/cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	"dubbo.apache.org/dubbo-go/v3/protocol/triple/internal/proto/greettriple"
+	"github.com/dubbogo/gost/log/logger"
+)
+
+func main() {
+	//client initialization
+	cli, err := client.NewClient(
+		client.WithURL("127.0.0.1:8080"),
+	)
+	dubboCli, err := greettriple.NewGreetServiceClient(cli)
```
Comment (chickenlj) [inline protocol/triple/internal/client/cmd/main.go:17]: Will `GreetService` be better than `GreetServiceClient`?

File: client/client.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	}
+	for _, opt := range opts {
+		opt(refCfg)
+	}
+	return &Client{
+		cfg: refCfg,
+	}, nil
+}
+
+func NewConsumerConfig(opts ...ConsumerOption) (*ConsumerConfig, error) {
```
Comment (chickenlj) [inline client/client.go:124]: I think it's better to expose only `NewClientXxx` varies in this package to users. Move `NewConsumerConfig` and `NewMethodConfig` to a new place or the previous place where it is from.

File: protocol/triple/internal/client/cmd/main.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	"dubbo.apache.org/dubbo-go/v3/client"
+	_ "dubbo.apache.org/dubbo-go/v3/imports"
+	greet "dubbo.apache.org/dubbo-go/v3/protocol/triple/internal/proto"
+	"dubbo.apache.org/dubbo-go/v3/protocol/triple/internal/proto/greettriple"
+	"github.com/dubbogo/gost/log/logger"
+)
+
+func main() {
+	//client initialization
+	cli, err := client.NewClient(
```
Comment (chickenlj) [inline protocol/triple/internal/client/cmd/main.go:14]: Is `dubbogo.yml` still needed to run with this API?

File: client/options.go [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+	perrors "github.com/pkg/errors"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+type CallOptions struct {
+	RequestTimeout string
+	Retries        string
+}
```
Comment (chickenlj) [inline client/options.go:16]: The options exposed here should be further investigated to make sure it's a 100% copy of the `config` package in regard of functionality.

**Review by chickenlj — CHANGES_REQUESTED** [review]

File: protocol/triple/internal/proto/greettriple/greet.triple.go [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+		return nil, err
+	}
+	return &GreetServiceClientImpl{
+		cli: cli,
+	}, nil
+}
+
+func SetConsumerService(srv common.RPCService) {
+	config.SetClientInfoService(&GreetService_ClientInfo, srv)
+}
```
Comment (chickenlj) [inline protocol/triple/internal/proto/greettriple/greet.triple.go:79]: How about we remove all the unnecessary old code stubs from the generated file?
For users who rely on old stubs, they can still use the old compiler to generate the code.

**Review by chickenlj — CHANGES_REQUESTED** [review]

File: protocol/triple/internal/client/cmd/main.go (hunk 1 above)
Comment (DMwangnima) [reply protocol/triple/internal/client/cmd/main.go:17]: Yes, NewGreetService would be better than NewGreetServiceClient cause two client concepts create confusion.

File: protocol/triple/internal/client/cmd/main.go (hunk 3 above)
Comment (DMwangnima) [reply protocol/triple/internal/client/cmd/main.go:14]: No, dubbogo.yml is only needed for classic configuration (SetConsumerService, config.Load). In this case, just create client with options.

File: protocol/triple/internal/proto/greettriple/greet.triple.go (hunk 5 above)
Comment (DMwangnima) [reply protocol/triple/internal/proto/greettriple/greet.triple.go:79]: The code generated by old compiler still depends on grpc library. Maybe there is a scenario that user wants to upgrade all services with new compiler? In this case, user also needs a classical config loading way to solve the complex configuration problem.(Assume that we have not provided NewClient with dubbogo.yml)

File: client/options.go (hunk 4 above)
Comment (DMwangnima) [reply client/options.go:16]: Yes, It's a placeholder that waits for professionals to think about what fields should be exposed.

**Pull request merged**